core/src/main/scala/kafka/controller/ReplicaStateMachine.scala

```diff
@@ -193,18 +232,38 @@ class ZkReplicaStateMachine(config: KafkaConfig,
           }
         }
       case OnlineReplica =>
+        var timings: AggregatedTimings = if (timingsOpt.isDefined) timingsOpt.get else AggregatedTimings()
```

Comment:
Instead of using a separate variable `timings`, I feel a more Scala-native way is to
change the following blocking of
if (timingsOpt.isDefined) {
...
}
to
timingsOpt.foreach { timings =>
...
}